lua/ulx_yeet/sh_yeet.lua

```diff
@@ -0,0 +1,173 @@
+end
+
+local function unRagdollPlayer( ragdoll )
+    if not IsValid( ragdoll ) then return end
+    local ply = ragdoll.player
+    if not IsValid( ply ) then return end
+    ply:SetParent()
+    ply:UnSpectate()
+    ply:Spawn()
+    restorePlayer( ply )
+
+    ply:SetPos( ragdoll:GetPos() )
+    ply:SetVelocity( ragdoll:GetVelocity() )
+    local yaw = ragdoll:GetAngles().yaw
+    ply:SetAngles( Angle( 0, yaw, 0 ) )
+    ragdoll:Remove()
+end
+
+local function ragdollPlayer( ply, velocity )
+    savePlayer( ply )
+
+    local ragdoll = ents.Create( "prop_ragdoll" )
+    if not IsValid( ragdoll ) then return end
+
+    ragdoll:SetModel( ply:GetModel() )
+    ragdoll:SetPos( ply:GetPos() )
+    ragdoll:SetAngles( ply:GetAngles() )
+    ragdoll:SetVelocity( velocity )
+    ragdoll:Spawn()
+
+    local boneCount = ragdoll:GetPhysicsObjectCount() - 1
+
+    for i = 0, boneCount do
+        local bonePhys = ragdoll:GetPhysicsObjectNum( i )
+        if IsValid( bonePhys ) then
+            local boneVec, boneAng = ply:GetBonePosition( ragdoll:TranslatePhysBoneToBone( i ) )
+            if boneVec and boneAng then
+                bonePhys:SetPos( boneVec )
+                bonePhys:SetAngles( boneAng )
+            end
+            bonePhys:SetVelocity( velocity )
+        end
+    end
+
+    ply:Spectate( OBS_MODE_CHASE )
+    ply:SpectateEntity( ragdoll )
+    ply:StripWeapons()
+
+    return ragdoll
+end
+
+
+local function playerPickup( ply, ent )
+    local access, tag = ULib.ucl.query( ply, "ulx physgunplayer" )
+    if ent:IsPlayer() and ULib.isSandbox() and access and not ent.NoNoclip and not ent.frozen and ply:GetInfoNum( "cl_pickupplayers", 1 ) == 1 then
+        local restrictions = {}
+        ULib.cmds.PlayerArg.processRestrictions( restrictions, ply, {}, tag and ULib.splitArgs( tag )[1] )
+        if restrictions.restrictedTargets == false or (restrictions.restrictedTargets and not table.HasValue( restrictions.restrictedTargets, ent )) then
+            return
+        end
+
+        if CLIENT then return true end
+
+        ent:SetMoveType( MOVETYPE_NONE )
+        local newPos = ent:GetPos()
+        local oldPos = ent:GetPos()
+        local steamId = ent:SteamID64()
+
+        hook.Add( "Tick", "CFC_Yeet_TickHolding_" .. steamId, function()
+            if not IsValid( ent ) then
+                hook.Remove( "Tick", "CFC_Yeet_Tick_Holding" .. steamId )
+                return
+            end
+
+            newPos = ent:GetPos()
+            ent.cfcYeetSpeed = newPos - oldPos
+            oldPos = ent:GetPos()
+        end)
+
+        return true
+    end
+end
+hook.Add( "PhysgunPickup", "ulxPlayerPickup", playerPickup, HOOK_HIGH )
+
+if CLIENT then return end
+
+local function playerDrop( ply, ent )
+    if not ent:IsPlayer() then return end
+
+    hook.Remove( "Tick", "CFC_Yeet_Tick_Holding" .. ent:SteamID64() )
+
+    ent:SetMoveType( MOVETYPE_WALK )
+    ent:SetVelocity( ent.cfcYeetSpeed * 50 )
```

Comment:
This calculation is used later, too, so we can just save it here

```suggestion
    local newVelocity = ent.cfcYeetSpeed * 50
    ent:SetMoveType( MOVETYPE_WALK )
    ent:SetVelocity( newVelocity )
```